src/excmds.ts

```diff
@@ -2664,7 +2666,191 @@ export async function viewcontainers() {
 // }}}
-//
+
+// {{{ TAB GROUPS
+/** @hidden */
+//#content_helper
+// {
+browser.tabs.onCreated.addListener(tgroupHandleTabCreated)
+browser.tabs.onRemoved.addListener(tgroupHandleTabRemoved)
+browser.tabs.onDetached.addListener(tgroupHandleTabDetached)
+browser.tabs.onAttached.addListener(tgroupHandleTabAttached)
+browser.tabs.onActivated.addListener(tgroupHandleTabActivated)
+browser.tabs.onUpdated.addListener(tgroupHandleTabUpdated)
+// }
+
+/**
+ * Create a new tab group in the current window.
+ *
+ * @param name The name of the tab group to create.
+ *
+ * If no tab groups exist, set the tab group name for all existing tabs in the
+ * window. Otherwise open a new tab and hide all tabs in the old tab group.
+ *
+ * Tab groups exist only for a single window.
+ *
+ */
+//#background
+export async function tgroupcreate(name: string) {
+    const promises = []
+    const groups = await tgroups()
+
+    if (groups.has(name)) {
+        throw new Error(`Tab group "${name}" already exists`)
+    }
+
+    if (groups.size > 0) {
+        await setWindowTgroup(name)
+        const initialUrl = await config.get("tabgroupnewtaburls")[name]
+        await tabopen(initialUrl)
+        promises.push(tgroupRun(name, true).then(tabs => browserBg.tabs.hide(tabs.map(tab => tab.id))))
+    } else {
+        promises.push(browser.tabs.query({currentWindow: true}).then(async (tabs) => {
+            const ids = tabs.map(tab => tab.id)
+            await setTabTgroup(name, ids)
+        }))
+        promises.push(setWindowTgroup(name))
+    }
+
+    groups.add(name)
+    promises.push(setTgroups(groups))
+    return Promise.all(promises).then(() => undefined)
+}
+
+/**
+ * Switch to a different tab group, hiding all other tabs.
+ *
+ * @param name The name of the tab group to switch to.
+ *
+ * If the tab group does not exist, act like tgroupcreate.
+ *
+ */
+//#background
+export async function tgroupswitch(name: string) {
+    if (name == await windowTgroup()) {
+        throw new Error(`Already on tab group "${name}"`)
+    }
+
+    const groups = await tgroups()
+    if (groups.size > 0) {
+        if (groups.has(name)) {
+            return tgroupActivate(name).then(() => undefined)
+        } else {
+            return tgroupcreate(name).then(() => undefined)
+        }
+    } else {
+        return tgroupcreate(name).then(() => undefined)
+    }
+}
+
+/**
+ * Switch to the previously active tab group.
+ */
+//#background
+export async function tgrouplast() {
+    if ((await tgroups()).size < 2) {
+        throw new Error("No last tab group")
+    }
+
+    return tgroupActivateLast().then(() => undefined)
+}
+
+/**
+ * Rename the current tab group.
+ *
+ * @param name The new name of the tab group.
+ *
+ */
+//#background
+export async function tgrouprename(name: string) {
+    if ((await tgroups()).size == 0) {
+        throw new Error("No tab groups exist")
+    }
+
+    return tgroupClearOldInfo(await windowTgroup() as string, name).then(() => undefined)
```

Comment:
```suggestion
    return tgroupClearOldInfo(await windowTgroup(), name).then(() => undefined)
```
If you put the cast in windowTgroup.